### ISSUE TITLE
Site name in app bar probably shouldn't update when editing until after Save is pressed

### ISSUE THREAD

**Issue by knipec** [techmatters/terraso-mobile-client#1658]
## Repro
1. Go to settings for a site
2. Edit the name
Expected: The site name in the green app bar at the top doesn't change until after you press Save
Actual: The site name appears to update in the app bar as you type. But it's not officially updating the real site name until you press Save. I find this kind of confusing because I'd expect the app bar site name to be the real site name.
https://github.com/techmatters/terraso-mobile-client/assets/5590815/c67d3eab-4e7e-41aa-a24d-e5c2824019ac

**Comment by paulschreiber** [comment]
The current behaviour is fine.

**Comment by DerekCaelin** [comment]
Curious about @CourtneyLee333's opinion about this issue

**Comment by CourtneyLee333** [comment]
I agree with Carissa's assessment. The name in the app bar should not update until Save.

**Comment by knipec** [comment]
@CourtneyLee333 @tm-ruxandra Do you consider this part of the components library work?

**Comment by tm-ruxandra** [comment]
@knipec I wouldn't, this seems like a bug.

**Comment by DerekCaelin** [comment]
Edited title and description, looking forward to triaging

**Comment by knipec** [comment]
Triage notes: 
- We think the desired behavior exists in the app for editing a project. Wonder if there's a difference in Formik usage. @knipec Is this an existing issue?

**Comment by knipec** [comment]
This should be fixed with work from https://github.com/techmatters/terraso-mobile-client/pull/3297/commits

**Comment by DerekCaelin** [comment]
Great! Will review in the next build.

**Comment by bjorwol** [comment]
Confirmed resolved. Tested on 1.4.7 staging, behaving correctly now.
